readme.md

```diff
@@ -79,28 +134,341 @@ The default export is `rehypeSanitize`.
+// doesn’t emit `hashchange`.
+document.addEventListener(
+  'click',
+  (event) => {
+    if (
+      event.target &&
+      event.target instanceof HTMLAnchorElement &&
+      event.target.href === location.href &&
+      location.hash.length > 1
+    ) {
+      setTimeout(() => {
+        if (!event.defaultPrevented) {
+          hashchange()
+        }
+      })
+    }
+  },
+  false
+)
+
+function hashchange() {
+  /** @type {string|undefined} */
+  let hash
+
+  try {
+    hash = decodeURIComponent(location.hash.slice(1)).toLowerCase()
+  } catch {
+    return
+  }
+
+  const name = 'user-content-' + hash
+  const target =
+    document.getElementById(name) || document.getElementsByName(name)[0]
+
+  if (target) {
+    setTimeout(() => {
+      target.scrollIntoView()
+    }, 0)
+  }
+}
+```
+
+### Example: math
+
+Math can be enabled in rehype by using the plugins
+[`rehype-katex`][rehype-katex] or [`rehype-mathjax`][rehype-mathjax].
+The operate on `span`s and `div`s with certain classes and inject complex markup
+and of inline styles, most of which this plugin will remove.
+Say our module `example.js` contains:
+
+```js
+import {unified} from 'unified'
+import rehypeParse from 'rehype-parse'
+import rehypeKatex from 'rehype-katex'
+import rehypeSanitize from 'rehype-sanitize'
+import rehypeStringify from 'rehype-stringify'
+
+main()
+
+async function main() {
+  const file = await unified()
+    .use(rehypeParse, {fragment: true})
+    .use(rehypeKatex)
+    .use(rehypeSanitize)
+    .use(rehypeStringify)
+    .process('<span class="math math-inline">L</span>')
+
+  console.log(String(file))
+}
+```
+
+Running that yields:
+
+```html
+<span><span><span>LL</span><span aria-hidden="true"><span><span></span><span>L</span></span></span></span></span>
+```
+
+It is possible to pass a schema which allows MathML and inline styles, but it
+would be complex, and allows *all* inline styles, which is unsafe.
+Alternatively, and arguably better, would be to *first* sanitize the HTML,
+allowing only the specific classes that `rehype-katex` and `rehype-mathjax` use,
+and *then* using those plugins:
+
+```diff
+@@ -1,7 +1,7 @@
+ import {unified} from 'unified'
+ import rehypeParse from 'rehype-parse'
+ import rehypeKatex from 'rehype-katex'
+-import rehypeSanitize from 'rehype-sanitize'
++import rehypeSanitize, {defaultSchema} from 'rehype-sanitize'
+ import rehypeStringify from 'rehype-stringify'
+
+ main()
+@@ -9,8 +9,21 @@ main()
+ async function main() {
+   const file = await unified()
+     .use(rehypeParse, {fragment: true})
++    .use(rehypeSanitize, {
++      ...defaultSchema,
++      attributes: {
++        ...defaultSchema.attributes,
++        div: [
++          ...(defaultSchema.attributes.div || []),
++          ['className', 'math', 'math-display']
++        ],
++        span: [
++          ...(defaultSchema.attributes.span || []),
++          ['className', 'math', 'math-inline']
++        ]
++      }
++    })
```

Comment:
This is a nice example as I comes directly from some issue, if I recall correctly